### PR TITLE
SIM2D-126: Work around unphysical detectorMap features

### PR DESCRIPTION
**Pull request by PaulPrice**

### CODE REVIEW

File: python/pfs/instmodel/splinedPsf.py [hunk 1]
```diff
@@ -1000,15 +1010,16 @@ def makeDetectorMap(self, fname, obsdate=None):
...
+        metadata.addString('CALIB_ID',
+                           'arm=%s spectrograph=%d ccd=%d filter=%s calibDate=%s calibTime=%s visit0=0' %
                            (self.detector.arm.value, self.detector.spectrograph, ccd,
-                            self.detector.arm.value, calibDate))
+                            self.detector.arm.value, calibDate, calibTime))
 
         detMap.writeFits(fname)
 
-        return detMap, fiberIds, allYKnots, allXCenters
+        return detMap
```
Comment (hassanxp) [inline python/pfs/instmodel/splinedPsf.py:1025]: So the return values other than ``detMap`` are never used in the code?

File: python/pfs/instmodel/splinedPsf.py [hunk 2]
```diff
@@ -972,25 +972,35 @@ def makeDetectorMap(self, fname, obsdate=None):
...
 
             yKnot = self._focalPlaneYToDetectorY(coeffs.ycCoeffs.get_coeffs(), correctToLL=True) / pixelScale
             xc = self._focalPlaneXToDetectorX(coeffs.xcCoeffs.get_coeffs(), correctToLL=True) / pixelScale
-            allYKnots.append(yKnot.astype(np.float32))
-            allXCenters.append(xc.astype(np.float32))
-
             assert len(lams) == len(yKnot)
             assert len(lams) == len(xc)
 
+            xCenterKnots.append(yKnot.astype(np.float32))
```
Comment (hassanxp) [inline python/pfs/instmodel/splinedPsf.py:987]: The ``xCenterKnots`` are constructed from ``yKnots``??

**Review by hassanxp — COMMENTED** [review]
Looks OK overall. Lots of flake8 violations across the entire file `splinedPsf.py` (mainly whitespaces). That may be fixed in a separate ticket.

File: python/pfs/instmodel/splinedPsf.py (hunk 2 above)
Comment (PaulPrice) [reply python/pfs/instmodel/splinedPsf.py:987]: Yes: the knots from one spline are used in another.

As for the style violations, I think that's falls under a cleanup ticket for the simulator that's never been scheduled.

**Pull request merged**